src/bar.cpp

```diff
@@ -607,6 +590,26 @@ void waybar::Bar::handleSignal(int signal) {
   }
 }
 
+auto waybar::Bar::setVisible(bool nvis) -> void {
+  visible = nvis;
+  if (!visible) {
+    window.get_style_context()->add_class("hidden");
+    surface_impl_->setLayer(bar_layer::BOTTOM);
+  } else {
+    window.get_style_context()->remove_class("hidden");
+    surface_impl_->setLayer(bar_layer::TOP);
```

Comment:
This overrides the layer set in the config file.
You can implement that differently:
```cpp
void waybar::Bar::setLayer(bar_layer layer) {
  layer_ = layer;
  surface_impl_->setLayer(layer_);
}
```
and call the method from the same location where you are disabling exclusive zone.